File: Day01/Medium/Solution.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
  public:
    void bubbleSort(vector<int>& arr) {
        int n = arr.size();
        for (int i = 0; i < n - 1; i++) {
            int flag = 0;
            for (int j = 1; j < n - i; j++) {
                if (arr[j] < arr[j - 1]) {
                    flag = 1;
                    swap(arr[j - 1], arr[j]);
                }
            }
            if (flag == 0) {
                return;
            }
        }
    }
};

int main() {
    vector<int> arr = {5, 1, 4, 2, 8};
    Solution sol;
    sol.bubbleSort(arr);

    cout << "Sorted array: ";
    for (int x : arr) cout << x << " ";
    cout << endl;
    return 0;
}

File: Day01/Medium/Solution.java
class Solution {
    public void bubbleSort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            int flag = 0;
            for (int j = 1; j < n - i; j++) {
                if (arr[j] < arr[j - 1]) {
                    flag = 1;
                    int temp = arr[j];
                    arr[j] = arr[j - 1];
                    arr[j - 1] = temp;
                }
            }
            if (flag == 0) {
                return;
            }
        }
    }

    public static void main(String[] args) {
        int[] arr = {5, 1, 4, 2, 8};
        Solution sol = new Solution();
        sol.bubbleSort(arr);

        System.out.print("Sorted array: ");
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
    }
}
